Backend/app/dashboard/rider-dashboard/src/Domain/Action/RiderPlatform/Management/Ride.hs

```diff
@@ -0,0 +1,100 @@
+    getRideRideInfo,
+    getRideTripRoute,
+    getRidePickupRoute,
+    postRideSyncMultiple,
+    postRideCancelMultiple,
+    getRideKaptureList,
+  )
+where
+
+import qualified API.Types.RiderPlatform.Management
+import qualified API.Types.RiderPlatform.Management.Ride hiding (BookingStatus)
+import qualified Dashboard.Common
+import qualified Dashboard.Common.Ride
+import qualified Dashboard.RiderPlatform.Management.Ride
+import qualified "lib-dashboard" Domain.Types.Merchant
+import qualified Domain.Types.Transaction
+import qualified "lib-dashboard" Environment
+import EulerHS.Prelude
+import qualified Kernel.External.Maps
+import qualified Kernel.Prelude
+import qualified Kernel.Types.APISuccess
+import qualified Kernel.Types.Beckn.Context
+import qualified Kernel.Types.Id
+import Kernel.Utils.Common
+import qualified RiderPlatformClient.RiderApp.Operations
+import qualified SharedLogic.Transaction
+import Storage.Beam.CommonInstances ()
+import Tools.Auth.Api
+import Tools.Auth.Merchant
+
+getRideList ::
+  Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant ->
+  Kernel.Types.Beckn.Context.City ->
+  ApiTokenInfo ->
+  Kernel.Prelude.Maybe (Kernel.Prelude.Int) ->
+  Kernel.Prelude.Maybe (Kernel.Prelude.Int) ->
+  Kernel.Prelude.Maybe (Dashboard.RiderPlatform.Management.Ride.BookingStatus) ->
+  Kernel.Prelude.Maybe ((Kernel.Types.Id.ShortId Dashboard.Common.Ride)) ->
+  Kernel.Prelude.Maybe (Kernel.Prelude.Text) ->
+  Kernel.Prelude.Maybe (Kernel.Prelude.Text) ->
+  Kernel.Prelude.Maybe (Kernel.Prelude.UTCTime) ->
+  Kernel.Prelude.Maybe (Kernel.Prelude.UTCTime) ->
+  Environment.Flow API.Types.RiderPlatform.Management.Ride.RideListRes
+getRideList merchantShortId opCity apiTokenInfo limit offset bookingStatus rideShortId customerPhoneNo driverPhoneNo from to = do
+  checkedMerchantId <- merchantCityAccessCheck merchantShortId apiTokenInfo.merchant.shortId opCity apiTokenInfo.city
+  RiderPlatformClient.RiderApp.Operations.callRiderAppOperations checkedMerchantId opCity (.rideDSL.getRideList) limit offset bookingStatus rideShortId customerPhoneNo driverPhoneNo from to
+
+getRideRideinfo ::
+  Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant ->
+  Kernel.Types.Beckn.Context.City ->
+  ApiTokenInfo ->
+  Kernel.Types.Id.Id Dashboard.Common.Ride ->
+  Environment.Flow API.Types.RiderPlatform.Management.Ride.RideInfoRes
+getRideRideinfo merchantShortId opCity apiTokenInfo rideId = do
+  checkedMerchantId <- merchantCityAccessCheck merchantShortId apiTokenInfo.merchant.shortId opCity apiTokenInfo.city
+  RiderPlatformClient.RiderApp.Operations.callRiderAppOperations checkedMerchantId opCity (.rideDSL.getRideRideinfo) rideId
+
+getRideInfo :: (Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant -> Kernel.Types.Beckn.Context.City -> ApiTokenInfo -> Kernel.Types.Id.Id Dashboard.Common.Ride -> Environment.Flow API.Types.RiderPlatform.Management.Ride.ShareRideInfoRes)
+getRideInfo merchantShortId opCity apiTokenInfo rideId = do
```

Comment:
this is old logic for this handler:

```
shareRideInfo ::
  ShortId DM.Merchant ->
  City.City ->
  Id Common.Ride ->
  FlowHandler Common.ShareRideInfoRes
shareRideInfo merchantShortId opCity rideId = withFlowHandlerAPI' $ do
  shareRideApiRateLimitOptions <- asks (.shareRideApiRateLimitOptions)
  checkSlidingWindowLimitWithOptions (rideInfoHitsCountKey $ getId rideId) shareRideApiRateLimitOptions
  checkedMerchantId <- merchantCityAccessCheck merchantShortId merchantShortId opCity opCity
  Client.callRiderAppOperations checkedMerchantId opCity (.rides.shareRideInfo) rideId
```

https://github.com/nammayatri/nammayatri/pull/8867/files#diff-5a9822055f838ab9549bf1c241992050198c6d3f0aaa90e1dee91a6d65bd6c30L104

you should edit manually  your autogenerated code to make logic unchanged

Reply:
if you need to use dsl apis with no auth, you can use this command:

nix flake lock --update-input namma-dsl --override-input namma-dsl github:nammayatri/namma-dsl/7ef5ab98877b69fbfd29c28aa53100cf3cdcd6d4

here is dsl changes https://github.com/nammayatri/namma-dsl/pull/109/commits

Reply:
then you can run generator:
`, run-generator  -- --apply-hint --skip-update`
`--skip-update` used here to prevent change namma-dsl commit back to actual main branch